frontend/src/components/TransactionsList.tsx

```diff
@@ -1,12 +1,129 @@
-import { DrumIcon } from "lucide-react";
+import { DownloadIcon, DrumIcon, MoreHorizontalIcon } from "lucide-react";
 import { useRef, useState } from "react";
 import { CustomPagination } from "src/components/CustomPagination";
 import EmptyState from "src/components/EmptyState";
 import Loading from "src/components/Loading";
 import TransactionItem from "src/components/TransactionItem";
-
+import { Button } from "src/components/ui/button";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "src/components/ui/dropdown-menu";
+import { UpgradeDialog } from "src/components/UpgradeDialog";
 import { LIST_TRANSACTIONS_LIMIT } from "src/constants";
+import { useAlbyMe } from "src/hooks/useAlbyMe";
 import { useTransactions } from "src/hooks/useTransactions";
+import { ListTransactionsResponse, Transaction } from "src/types";
+import { request } from "src/utils/request";
+
+const convertToCSV = (transactions: Transaction[]) => {
+  if (!transactions.length) {
+    return "";
+  }
+
+  // Get headers from the first transaction
+  const headers = Object.keys(transactions[0]);
+  const csvHeaders = headers.join(",");
+
+  // Convert each transaction to CSV row
+  const csvRows = transactions.map((tx) => {
```

Comment:
Here we do csv generation manually, isn't there a library we can use? I think this will be safer to ensure we don't make a mistake with the encoding